### DISCUSSION TITLE
How to implement Navbar with collapsed icons

### DISCUSSION THREAD

**Discussion by lukasbash** [mantinedev/mantine#5169] [Q&A] [answered]
Doesn't it make sense to provide an additional mode in the navbar (similar to the collapsed property) to make only icons visible instead of fully collapsing it?
I saw the sectioned navigation example at MantineUI, but it is simply a lot of overhead to handle all the styles, animations and breakpoints on its own.
Possible to implement this as a near feature?

**Comment by rtivital** [comment, marked as answer]
AppShell is a layout component. It is not planned to introduce these features because the structure varies a lot. This is a part of application logic, not part of the library.

**Reply by ianpavlovski** [reply]
I disagree with the Mantine team’s position on this issue. This is indeed an incorrect component design philosophy

The “icon-only” mode is a basic feature of any modern navigation bar. Look at any popular UI framework:
	•	Material UI provides full solutions with a responsive drawer that can collapse to icon-only mode
	•	Bootstrap has built-in support for collapsed states while preserving important elements
	•	Flowbite demonstrates a sticky navbar with adaptive behavior

All these solutions recognize that collapsing to icons is not “custom application logic” but a fundamental need of modern interfaces.

If Mantine positions AppShell as a universal layout solution, the team should bring it to logical completion. You cannot create a “universal” component and then refuse to implement its basic functions under the pretext that “this is application logic.”

This creates a situation where:
	•	Developers are forced to reinvent the wheel for standard functionality
	•	Consistency between projects is lost
	•	Development time increases to solve already solved problems

Instead of refusing functionality, the team could provide optional modes:

```
<AppShell.Navbar 
    collapsed={{ desktop: false, mobile: true }}
    iconOnly={{ desktop: false, mobile: true }}
    width={{ base: 280, iconOnly: 60 }}
>
```

Or at least official examples with ready-made styles and animations for such behavior so that developers do not start from scratch.

So refusing to implement standard functionality under the pretext of “architectural purity” is a pseudo-approach that practically complicates developers’ lives. If AppShell claims to be a full-fledged layout solution, it should cover typical use cases and not force each developer to solve the same tasks independently.

**Reply by btroop** [reply]
agreed @lukasbash. this sort of functionality seems pretty fundamental to what the appshell/mantine is trying to achieve
